src/main/resources/org/jenkinsci/plugins/scriptsecurity/sandbox/whitelists/generic-whitelist

```diff
@@ -34,13 +34,31 @@ method groovy.lang.Closure setResolveStrategy int
 method groovy.lang.GString plus java.lang.String
 method groovy.lang.Script getBinding
 
+# Useful to show stacktrace to the user.
+new java.io.PrintWriter java.io.Writer
 # Needed for Groovy Templates to emit output:
 method java.io.PrintWriter print java.lang.Object
 method java.io.PrintWriter print java.lang.String
 
 # Useful general stuff:
 new java.io.StringReader java.lang.String
+method java.io.StringReader close
+method java.io.StringReader read
```

Comment:
You rather want to whitelist the _defining_ method, in `Reader`. Ditto in other cases.

Reply:
Missed to respond to this earlier. That would open up those methods for all readers, right? Won't that whitelist too much, e.g., reading local files etc.?

Reply:
Looks like specific overridden methods can't be whitelisted, so it is all or nothing.

Reply:
I pushed a new change with the right attribution. I am assuming it is still OK because as long as the sandbox prevents the construction of those objects, there is not going to be a security issue with the broader whitelisting.

Reply:
> reading local files

You would first need to construct a `FileReader`, which would not be allowed.